### PR TITLE
templates: bunch of small cleanups

### PR DESCRIPTION
**Pull request by prince-chrismc**
Docs!

I did quick pass to clean things up -- this release is not available but some of the changes are backwards.

New feature can be added as comments since it's not available in CCI

### CODE REVIEW

File: docs/package_templates/autotools_package/all/conanfile.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def build(self):
     def package(self):
         copy(self, pattern="LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
         autotools = Autotools(self)
-        # TODO: replace by autotools.install() once https://github.com/conan-io/conan/issues/12153 fixed
+        # TODO: replace by autotools.install() once Conan 1.54 is available in CCI
```
Comment (SpaceIm) [inline docs/package_templates/autotools_package/all/conanfile.py:160]:
```suggestion
        # TODO: replace by autotools.install() if required_conan_version greater or equal than 1.54
```

**Review by jwillikers — CHANGES_REQUESTED** [review]
@SpaceIm Has a good suggestion.

File: docs/package_templates/autotools_package/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply docs/package_templates/autotools_package/all/conanfile.py:160]: The templates should generally just be the latest. As per other convo - we dont except contributors to know the different version. If they do that's amazing ❤️ 

But I'd rather not confuse any who doesn't know what version is available in CCI versus what they can install locally :)

For that I will not be apply the suggestion - hopefully that makes sense

Comment (SpaceIm) [reply docs/package_templates/autotools_package/all/conanfile.py:160]: I don't understand your reasoning.

Comment (prince-chrismc) [reply docs/package_templates/autotools_package/all/conanfile.py:160]: I am afraid  Contributors won't understand you suggestion.. most are aware of the difference between what Conan release, what CCI is build and what the templates are written for

If that assertion is wrong its not our expectation or a requirement for them too and the docs need to be written as such

**Review by uilianries — APPROVED** [review]

**Pull request merged**